tests/CMakeLists.txt

```
@@ -182,12 +181,37 @@ function(add_cts_test)
   target_sources(test_all PRIVATE $<TARGET_OBJECTS:${test_exe_name}_objects>)
 endfunction()
 
-file(GLOB test_category_dirs RELATIVE ${CMAKE_CURRENT_SOURCE_DIR} *)
-list(REMOVE_ITEM test_category_dirs "common")
-foreach(dir ${test_category_dirs})
-  if(EXISTS "${CMAKE_CURRENT_SOURCE_DIR}/${dir}/CMakeLists.txt")
-    add_subdirectory(${dir})
+function(add_cts_test)
+  # To make check that any .cpp files are passed
+  # List created because, direct check on "${ARGN}" gives false-positive result
+  set(tests_list "${ARGN}")
+  if (tests_list)
+    get_filename_component(test_dir ${CMAKE_CURRENT_SOURCE_DIR} NAME)
+    set(test_exe_name ${test_dir})
+    set(test_cases_list "${ARGN}")
+
+    add_cts_test_helper(${test_exe_name} "${test_cases_list}")
   endif()
+endfunction()
+
+function(add_independent_cts_tests)
+  set(tests_list "${ARGN}")
+  foreach(ind_test IN LISTS tests_list)
+    if(EXISTS "${ind_test}")
+      get_filename_component(cpp_name "${ind_test}" NAME_WE)
+      set(test_exe_name "${cpp_name}")
+      set(test_cases_list "${ind_test}")
+
+      add_cts_test_helper(${test_exe_name} "${test_cases_list}")
+    else()
+      # this comment will be removed after testing TMT run
+      message(STATUS ">>>>>>>>>>>>>>>>>>>>> no file named ${ind_test}")
```

```suggestion
      # ...
```
There should be an explanation for why it's OK to skip a file that doesn't exist

---

now, generating CMake files process will be stopped, if at least one file doesn't exist
I'll do it in this commit: 21f8fc9e15876fd4e9c2de23deb84e3327021dd0